Lib/test/test_zipfile.py

```diff
@@ -2234,6 +2236,90 @@ def test_create_command(self):
             finally:
                 unlink(TESTFN2)
 
+    @requires_zlib
+    def test_create_wcard_command(self):
+        files = []
+        extlist = ['a', 'aa', 'ab', 'abc', 'b', 'bb', 'c', 'cd', 'cde']
+        for ext in extlist:
+            fn = "{}.{}".format(TESTFN, ext)
+            with open(fn, 'w') as f:
+                f.write('test {}'.format(ext))
+                self.addCleanup(unlink, fn)
+                files.append(fn)
+        opt = '-c'
+        for pat, expect in [('{}.?', 3), ('{}.a*', 4), ('{}.???', 2), ('{}.[ab]*', 6), ]:
+            try:
+                out = self.zipfilecmd(opt, TESTFN2, pat.format(TESTFN))
+                #self.assertEqual(out, b'')
+                with zipfile.ZipFile(TESTFN2) as zf:
+                    self.assertEqual(len(zf.namelist()), expect)
```

Comment:
There are other tests that are testing that files have the correct content so I felt that length was sufficient. Have given a more detailed assert message.